docs/proposals/20220927-label-sync-between-machine-and-nodes.md

```diff
@@ -21,28 +25,28 @@ Refer to the [Cluster API Book Glossary](https://cluster-api.sigs.k8s.io/referen
 
 ## Summary
 
-This document discusses a means by which labels placed on Machine can be kept in sync with corresponding labels on the underlying Kubernetes Nodes.
+This document discusses how labels placed on a Machine can kept in sync with the corresponding Kubernetes Node.
 
 ## Motivation
 
-Managing labels on Kubernetes nodes is a long standing [issue](https://github.com/kubernetes-sigs/cluster-api/issues/493) in Cluster API.
+Managing labels on Kubernetes nodes has been a long standing [issue](https://github.com/kubernetes-sigs/cluster-api/issues/493) in Cluster API.
 
-During [previous attempt](https://github.com/kubernetes-sigs/cluster-api/pull/6255) to address this issue (thanks @Arvinderpal for all the work) we discovered that this topic has several implications:
+The following challenges have been identified through various iterations:
 
-- Define how labels propagate from Machine to node
-- Define how labels propagate from ClusterClass to KCP/MachineDeployments/Machine Pools and down to Machines
-- Define how to prevent that label propagation triggers un-necessary rollouts
+- Define how labels propagate from Machine to node.
+- Define how labels propagate from ClusterClass to KubeadmControlPlane/MachineDeployments/Machine Pools, and ultimately to Machines.
+- Define how to prevent that label propagation triggers unnecessary rollouts.
 
 With the "divide and conquer" principle in mind this proposal aims to address the first point only, while the remaining points are going to be addressed in separated, complementary efforts.
 
 ### Goals
 
-- Support label synchronizing from Machine to the corresponding Kunernetes nodes for the labels that fall under the Kubernetes `node-role.kubernetes.io/` and `node-restriction.kubernetes.io/` prefix
-- Support label synchronizing from Machine to the corresponding Kunernetes nodes for the labels that fall under the Cluster API `node.cluster.x-k8s.io/` prefix
+- Support label sync from Machine to the linked Kubernetes node, limited to `node-role.kubernetes.io/` prefix and the `node-restriction.kubernetes.io` domain.
+- Support syncing labels from Machine to the linked Kubernetes node for the Cluster API owned `node.cluster.x-k8s.io` domain.
```

Comment:
For these ones, support authoritative sync?

Reply:
being authoritative means other controllers are not allowed to set any node restriction on the node; I'm not sure how many tools are doing this, but If we go we go down the authoritative path I think we will discover this the hard way.
I would prefer to avoid this...

Reply:
As mentioned elsewhere, I think the SSA-way of being authoritative would be ideal, aka being authoritative for all the labels we want to set now or we did set on the previous reconcile.

Reply:
In this comment I was referring to `cluster.x-k8s.io`
In this [other comment](https://github.com/kubernetes-sigs/cluster-api/pull/7296#discussion_r984442648) I was referring to the other ones.

Reply:
Ah I see. So the question is if we want to drop other labels in the `node.cluster.x-k8s.io` domain from the Node that are not set on the Machine. I.e. CAPI / the machine controller would be the only ones setting labels in that domain.

Sounds good to me, but no strong opinion from my side

Reply:
I think that if we go down the path to implement SSA, then we should keep everything consistent and manage only the labels we own (I won't make a special case for making `node.cluster.x-k8s.io` authoritative, it seems unnecessary complication).

Reply:
Agree